x/ecocredit/server/core/msg_create_class.go

```diff
@@ -143,28 +154,22 @@ func (k Keeper) CreateClass(goCtx context.Context, req *core.MsgCreateClass) (*c
 	return &core.MsgCreateClassResponse{ClassId: classID}, nil
 }
 
-func (k Keeper) assertCanCreateClass(sdkCtx sdk.Context, adminAddress sdk.AccAddress) error {
-	var allowListEnabled bool
-	k.paramsKeeper.Get(sdkCtx, core.KeyAllowlistEnabled, &allowListEnabled)
-	if allowListEnabled {
-		var allowList []string
-		k.paramsKeeper.Get(sdkCtx, core.KeyAllowedClassCreators, &allowList)
-		if !k.isCreatorAllowListed(sdkCtx, allowList, adminAddress) {
-			return sdkerrors.ErrUnauthorized.Wrapf("%s is not allowed to create credit classes", adminAddress.String())
-		}
+func (k Keeper) assertCanCreateClass(ctx context.Context, adminAddress sdk.AccAddress) error {
+	allowListEnabled, err := k.stateStore.AllowListEnabledTable().Get(ctx)
+	if err != nil {
+		return err
 	}
-	return nil
-}
 
-func (k Keeper) isCreatorAllowListed(ctx sdk.Context, allowlist []string, designer sdk.AccAddress) bool {
-	for _, addr := range allowlist {
-		allowListedAddr, _ := sdk.AccAddressFromBech32(addr)
-		if designer.Equals(allowListedAddr) {
-			return true
+	if allowListEnabled.Enabled {
+		_, err := k.stateStore.AllowedClassCreatorTable().Get(ctx, adminAddress)
+		if err != nil {
+			if ormerrors.NotFound.Is(err) {
+				return sdkerrors.ErrUnauthorized.Wrapf("%s is not allowed to create credit classes", adminAddress.String())
+			}
+			return err
 		}
```

Comment:
we could use `Has` here and not have to check the error type, but ok to leave as is